### PR TITLE
Timer fixes for auto-renew

### PR DESCRIPTION
**Pull request by nsharma123**
https://webarchive.jira.com/browse/WEBDEV-6894

### CODE REVIEW

File: src/ia-book-actions.js [hunk 1]
```diff
@@ -129,10 +128,19 @@ export default class IABookActions extends LitElement {
...
     this.disconnectResizeObserver();
   }
 
+  /**
+   * send log messages to sentry
+   * @param {string} msg
+   */
+  sentryCaptureMsg(msg) {
+    log(window?.Sentry);
+    if (window?.Sentry) window?.Sentry?.captureMessage(msg);
```
Comment (iisa) [inline src/ia-book-actions.js:141]: q: do we need conditional if statement has optional chain? can we remove conditional?

File: src/ia-book-actions.js [hunk 2]
```diff
@@ -283,6 +291,38 @@ export default class IABookActions extends LitElement {
...
+        );
+
+        if (
+          this.borrowType === 'browsed' &&
+          this.lendingStatus.browsingExpired === false
+        ) {
+          const loanTime = await this.localCache.get(
+            `${this.identifier}-loanTime`
+          );
+          const secondsLeft = Math.round((loanTime - new Date()) / 1000); // different in seconds
```
Comment (iisa) [inline src/ia-book-actions.js:315]: q: what does `different in seconds` comment mean? can we clarify and move to above this var declaration?

File: src/ia-book-actions.js [hunk 3]
```diff
@@ -487,36 +527,6 @@ export default class IABookActions extends LitElement {
     this.toast?.remove();
   }
 
-  /**
-   * @deprecated
```
Comment (iisa) [inline src/ia-book-actions.js:491]: ty

File: src/ia-book-actions.js [hunk 4]
```diff
@@ -677,55 +687,77 @@ export default class IABookActions extends LitElement {
...
-        resync,
+  /**
+   * setInterval execution function do following things
+   * - if setInterval timer gone off, let's resync
+   * - loan-renew attempt to renew the loan OR show relavant message
+   * - if loan has expired, clear interval timers
+   *
+   * @param {Number} secondsLeftOnLoan
+   */
+  async setIntervalExecutionBlock(secondsLeftOnLoan) {
```
Comment (iisa) [inline src/ia-book-actions.js:707]: let's describe this function's behaviour in the name, like, "loanStatusCheckInterval" 😄

File: src/ia-book-actions.js [hunk 5]
```diff
@@ -677,55 +687,77 @@ export default class IABookActions extends LitElement {
...
+    );
 
-      // clear interval in secondsLeft if less
-      if (secondsLeft <= this.timerExecutionSeconds) {
-        this.disconnectedCallback();
-        window?.Sentry?.captureMessage(sentryLogs.clearOneHourTimer);
-      }
-    }, this.timerExecutionSeconds * 1000);
+    /**
+     * execute from last 10th minute to 0th minute
```
Comment (iisa) [inline src/ia-book-actions.js:740]:
```suggestion
     * execute from last 10th minutes to 0th minute
```

File: test/ia-book-actions.test.js [hunk 6]
```diff
@@ -421,7 +421,7 @@ describe('Auto renew one hour loan', () => {
     // timer-countdown reflects decreased loan time
     expect(el.timerCountdownEl.secondsLeftOnLoan).to.equal(8);
 
-    expect(el.postInitComplete).to.equal(33);
+    expect(el.postInitComplete).to.equal(false);
```
Comment (iisa) [inline test/ia-book-actions.test.js:424]: q: why did we change this?

File: test/ia-book-actions.test.js [hunk 7]
```diff
@@ -445,7 +445,7 @@ describe('Auto renew one hour loan', () => {
     expect(el.loanRenewResult.renewNow).to.equal(true);
     expect(el.loanRenewResult.secondsLeft).to.equal(10);
     expect(el.timerCountdownEl).to.exist;
-    expect(el.timerCountdownEl.secondsLeftOnLoan).to.equal(loanTime);
+    expect(el.timerCountdownEl.secondsLeftOnLoan).to.equal(8);
```
Comment (iisa) [inline test/ia-book-actions.test.js:448]: Q: why did we change this?

**Review by iisa — COMMENTED** [review]
looks ok, Q: can we add unit test to visibilitychange event handler?

File: test/ia-book-actions.test.js (hunk 6 above)
Comment (phpdevonj) [reply test/ia-book-actions.test.js:424]: its a boolean property which considered as false after during execution.

Comment (nsharma123) [reply test/ia-book-actions.test.js:424]: Its a boolean property which expected to be false while execution.

File: test/ia-book-actions.test.js (hunk 7 above)
Comment (nsharma123) [reply test/ia-book-actions.test.js:448]: we have changed the secondsLeftOnLoan key when loan was renewed so new updated time needs to be there.

Comment (nsharma123) [reply test/ia-book-actions.test.js:449]: Yes

**Review by iisa — APPROVED** [review]
ok, lets merge and push to latest review app

**Pull request merged**